lib/client.js

```diff
@@ -130,6 +132,13 @@ Client.prototype.request = function(opts, res) {
         req.write(opts.body);
     req.end();
 
+    if (this.config.timeout){
+        req.setTimeout(this.config.timeout, function() {
+          req.abort();
+          res(new Error('connection timed out'));
```

Comment:
I honestly forget why I put this line in, most likely because I didn't want to lose root cause? Anyway, I agree 'error' event handler alone is enough, since it includes ECONNRESET code already.